debian/tests/integrationtests

```diff
@@ -1,29 +1,30 @@
 #!/bin/sh
 
-NEEDS_REBOOT=/tmp/needs-reboot
-
 set -ex
 
 # for these tests, run snap and snapd from outside of the core snap
-sudo mkdir -p /etc/systemd/system/snapd.service.d/
-cat <<EOF | sudo tee /etc/systemd/system/snapd.service.d/no-reexec.conf
+mkdir -p /etc/systemd/system/snapd.service.d/
+cat <<EOF | tee /etc/systemd/system/snapd.service.d/no-reexec.conf
 [Service]
 Environment=SNAP_REEXEC=0
 EOF
 
 # required for the debian adt host
 if [ "$http_proxy" != "" ]; then
```

Comment:
I guess we still need the http_proxy set up for executing autopkgtest in prodstack (it seems to be set in [1] for instance), if it's only because of our scalingstack CI infra we could get rid of it.

[1] https://objectstorage.prodstack4-5.canonical.com/v1/AUTH_77e2ada1e7a84929a74ba3b87153c0ac/autopkgtest-xenial/xenial/amd64/s/snapd/20160802_091607@/log.gz